grails-app/services/com/unifina/service/CanvasService.groovy

```diff
@@ -201,8 +203,47 @@ class CanvasService {
 		return grailsLinkGenerator.link(controller: 'canvas', action: 'editor', id: canvas.id, absolute: true)
 	}
 
+	@CompileStatic
+	def addExampleCanvases(SecUser user, List<Canvas> examples) {
+		for (final Canvas example : examples) {
+			switch (example.exampleType) {
+				// Create a copy of the example canvas for the user and grant read/write/share permissions.
+				case ExampleType.COPY:
+					Canvas c = new Canvas()
+					InvokerHelper.setProperties(c, example.properties)
+					c.dashboardItems = new HashSet<DashboardItem>()
+					c.permissions = new HashSet<Permission>()
+					c.runner = null
+					c.server = null
+					c.requestUrl = null
+					c.serialization = null
+					c.startedBy = null
+					c.state = Canvas.State.STOPPED
+					c.exampleType = ExampleType.NOT_SET
+					c.save(validate: true, failOnError: true)
```

Comment:
Should save be called after updateExisting(...)?

Reply:
Or is calling save() here useful at all?

Reply:
`updateExisting` saves the canvas AND grants the user all permissions to it if it is a previously unsaved canvas. So for `ExampleType.COPY` it should be enough to setup the `Canvas` object and `SaveCanvasCommand` object, and call `updateExisting` - nothing else.